xmodule/tests/test_library_content.py

```diff
@@ -71,16 +87,59 @@ def get_block(descriptor):
         block.runtime.get_block_for_descriptor = get_block
 
 
+@ddt.ddt
+class LibraryContentGeneralTest(LibraryContentTest):
+    """
+    Test the base functionality of the LibraryContentBlock.
+    """
+
+    @ddt.data(
+        ('library-v1:ProblemX+PR0B', LibraryLocator),
+        ('lib:ORG:test-1', LibraryLocatorV2)
+    )
+    @ddt.unpack
+    def test_source_library_key(self, library_key, expected_locator_type):
+        """
+        Test the source_library_key property of the xblock.
+
+        The method should correctly work either with V1 or V2 libraries.
+        """
+        library = self.make_block(
+            "library_content",
+            self.vertical,
+            max_count=1,
+            source_library_id=library_key
+        )
+        assert isinstance(library.source_library_key, expected_locator_type)
+
+    def test_initial_sync_from_library(self):
+        """
+        Test that a lc block starts without children, but is correctly populated upon first sync.
+        """
+        source_library_key = self.library.location.library_key
+
+        # Normally the children get added when the "source_libraries" setting
+        # is updated, but the way we do it through a factory doesn't do that.
+        assert self.lc_block.source_library_key == source_library_key
+        assert self.lc_block.source_library_version is None
+        assert len(self.lc_block.children) == 0
+
+        # Update the LibraryContent block's children:
+        self._sync_lc_block_from_library()
+
+        # Check that all blocks from the library are now children of the block:
+        assert self.lc_block.source_library_key == source_library_key  # Unchanged
+        assert self.lc_block.source_library_version == self.tools.get_latest_library_version(source_library_key)
+        assert len(self.lc_block.children) == len(self.lib_blocks)
```

Comment:
A new test I added after struggling with bugs around the "uninitialized" edge case. This case is relevant when an LC block's library has been set, but the celery task to sync its children hasn't finished yet.